lib/process-file.js

```diff
@@ -48,45 +46,47 @@ module.exports = function processFile(inFile, outFile, relativePath, outDir, sil
 
   let start = new Date();
   debug('[starting]: %s %dKB', relativePath, (src.length / 1000));
-  let result = terser.minify(src, options);
-  let end = new Date();
-  let total = end - start;
-  if (total > 20000 && !silent) {
-    console.warn(`[WARN] (broccoli-uglify-sourcemap) Minifying "${relativePath}" took: ${total}ms (more than 20,000ms)`);
-  }
 
-  if (result.error) {
-    result.error.filename = relativePath;
-    throw result.error;
-  }
-
-  debug('[finished]: %s %dKB in %dms', relativePath, (result.code.length / 1000), total);
-
-  if (options.sourceMap) {
-    let newSourceMap = JSON.parse(result.map);
+  return terser.minify(src, options)
+    .then(result => {
```

Comment:
Let's make this an async function, then we can use async/await + try/catch here instead

Reply:
This package supports node 6 till now. `async/await` will require at least node version 8. Should the node 6 support be dropped with this change?

Reply:
We dropped Node 6 over in #241, so once you rebase this one it should be good to use `async`/`await`.

Reply:
done 👍🏽